src/psyclone/configuration.py

```diff
@@ -574,7 +606,106 @@ def get_default_keys(self):
         return self._config.defaults()
 
 
-class DynConfig(object):
+# =============================================================================
+class APISpecificConfig(object):
+    '''A base class for functions that each API-specific class must provide.
+    At the moment that is a function access_mapping().
+    :param section: :py:class:`configparser.SectionProxy`
+    :Raises ConfigurationError: if an access-mapping is provided that
+        assigns an invalid value (i.e. not one of 'read', 'write',
+        'readwrite'), 'inc' or 'sum') to a string.
+    '''
+
+    def __init__(self, section):
+        # Set a default mapping, this way the test cases all work without
+        # having to specify those mappings.
+        self._access_mapping = {"read": "read", "write": "write",
+                                "readwrite": "readwrite", "inc": "inc",
+                                "sum": "sum"}
+        # Get the mapping and convert it into a directory. The input is in
+        # the format: key1:value1, key2=value2, ...
+        mapping = section.get("ACCESS_MAPPING")
+        if mapping:
+            self._access_mapping = \
+                APISpecificConfig.create_dict_from_string(mapping)
+        # Now convert the string type ("read" etc) to AccessType
+        from psyclone.core.access_type import AccessType
+        for api_access_name, access_type in self._access_mapping.items():
+            try:
+                self._access_mapping[api_access_name] = \
+                    AccessType.from_string(access_type)
+            except ValueError:
+                # Raised by from_string()
+                raise ConfigurationError("Unknown access type '{0}' found "
+                                         "for key '{1}'"
+                                         .format(access_type,
+                                                 api_access_name))
+
+        # Now create the reverse lookup (for better error messages):
+        self._reverse_access_mapping = {v: k for k, v in
+                                        self._access_mapping.items()}
+
+    @staticmethod
+    def create_dict_from_string(input_str):
+        '''Takes an input string in the format:
+        key1:value1, key2:value2, ...
+        and creates a dictionary with the key,value pairs.
+        Spaces are removed.
+        :param str input_str: The input string.
+        :returns: A dictionary with the key,value pairs from the input string.
+        :rtype: dict.
+        :Raises ConfigurationError if the input string contains an entry
+                that is does not have a ":".
```

Comment:
Done.

Reply:
Almost! You need a colon after "ConfigurationError" :-)

Reply:
Gee, that was embarrassing (then again ... good case for being able to create an API guide, since it would allow a dev to easily check the correctness of markups). Done.